### DISCUSSION TITLE
How to add "exotic color space" support to CSS?

### DISCUSSION THREAD

**Discussion by SoftMoonWebWare** [color-js/color.js#578] [Q&A] [answered]
I swear I saw, in this repo, talk about adding "exotic" color-space support using JavaScript  for the CSS `color()`  function.  I just can't find it, not here, not at colorjs.io, not on the Mozilla Developer's Network, nowhere.

I really could use OKHSV in my project.  And then .... I have to hope browsers can create gradients through this exotic color-space.  Any current word on that possibility?

Otherwise, I'll have to create it manually with an HTML canvas.  HMMM....maybe I should anyway for when the project moves beyond sRGB....?  Either way, knowing how to add "exotic" color-space support to CSS would be nice.

**Comment by facelessuser** [comment]
Okhsv has already been added on main, it just hasn't been released formally yet.

I think you can pull the unreleased code from `import Color from "https://colorjs.io/dist/color.js";` or maybe `https://colorjs.io/src/color.js` Documentation doesn't make it clear whether `dist` or `src` should be used. Either way, I think you can pull the latest that way.

**Comment by SoftMoonWebWare** [comment]
That's a long file, and I don't have time here at the end of the day to study it.  It will be 2 weeks before I have more time.  Keeping current.....

So you're saying, in this file, I can find the code to add OKHSV support directly to CSS in an HTML page via JavaScript?
Could I find it in the GitHub repo then?  Isn't that file a "build" of that repo for us "old-school" tinkerers who don't use modules?  I'll look again...

Then I just hope browsers can grade "through" OKHSV with that support...

Y'all really are awesome.  Thanks again for your help!

Aloha!

**Reply by facelessuser** [reply]
I was more just saying that Okhsv support is available if you import that code. I'm not saying you need to look through all of it.

**Reply by SoftMoonWebWare** [reply]
Thanks for your prompt reply.
I'm interested in ***HOW*** to add exotic color-space support to CSS files, not just doing it.
My project is free of all external dependencies, and I intend to keep it that way.  Besides, that file is a ***LOT*** of overkill; if I understand it (without actually looking through all of it !!!) that file contains the entire project, all the color-spaces, etc.  My project already has the code for color-space conversions, tailored to my specific needs used throughout 6800+ lines of code developed over the last 12-13 years.  I would have to basically rewrite the logic to accommodate a different API if I were to use your project, and I'm one guy on budgeted time; and I doubt I would get the same performance if I used your project.  
Your API is simple, straightforward, easy to use, easy to manage, easy to understand, easy to remember, and more complete than mine all in one package.  Mine is complicated, and even I have to look at the code or the instructions page when I use it sometimes, and I wrote it, rewrote it, rewrote it again, and again, and again, with lots of updates in-between re-writes, so I should know it well!  But it does exactly what I need it to under various conditions, as efficiently as I can figure out how to code it.  I got a roughly 700% increase in performance at one point, skipping OO colors and fancy user-input filtering schemes, and sticking to native Arrays and strict input requirements.  The end-user went from 7-8 seconds of wait time for an HTML canvas to be developed to 1-2 seconds.  Yet I can configure it to use that convenient fancy API I/O scheme with OO colors that stringify as CSS specs and filtered user-input if I need.
I took today off from reality, unexpectedly, so I have some time to look through it.  I picked up a few coding tricks and got an "update on the state of JavaScript" from looking though y'all's "main" on GitHub, so I'm looking forward to it ☻!  Y'all's code made my code better, not just updated and expanded with new color-spaces.
Thanks again!
Aloha!

**Reply by facelessuser** [reply, marked as answer]
CSS does not directly support exotic color spaces. CSS color level 5 mentions custom color spaces that you can link to color profiles though: https://drafts.csswg.org/css-color-5/#custom-color. I don't know if any browsers support it though, and I think it is limited to RGB style profiles, but that's the closest you have.

The only other way to support exotic color spaces is to use a library that manipulates colors in the space you care about and then convert them to CSS supported color spaces.

**Reply by SoftMoonWebWare** [reply]
That is specifically what I needed to know, and specifically the link I could not find (my Google foo sucks).
Then I must create a "fully-chormatic∝" rainbow for the "OK" color space hues using an HTML canvas and OKHSV.
I was hoping to use a CSS gradient "through" OKHSV.
Ever make it West of the Front Range, I owe you about 10 cups of coffee!
**Again, thanks again!**
Aloha!
